### PR TITLE
Instrument test exporter for OpenTracing.

### PR DESCRIPTION
**Pull request by tomwilkie**
Also:
- Move `SpanLogger` out of `chunk` package and reuse it.
- Update the logging in the test-exporter to use gokit.

~Vendor modified in place, don't merge until https://github.com/prometheus/client_golang/pull/469/files is in.~

Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by tomwilkie** [comment]
Can do this with no upstream modifications.  Updated, read for review.

File: pkg/querier/correctness/runner.go [hunk 1]
```diff
@@ -125,6 +133,16 @@ func NewRunner(cfg RunnerConfig) (*Runner, error) {
 	return tc, nil
 }
 
+type tracingClient struct {
```
Comment (csmarchbanks) [inline pkg/querier/correctness/runner.go]: Is this used anywhere?

File: pkg/querier/correctness/runner.go [hunk 2]
```diff
@@ -105,11 +110,14 @@ func NewRunner(cfg RunnerConfig) (*Runner, error) {
 		Address: cfg.prometheusAddr,
 	}
 	if cfg.userID != "" {
-		apiCfg.RoundTripper = promhttp.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
-			user.InjectOrgIDIntoHTTPRequest(user.InjectOrgID(context.Background(), cfg.userID), req)
-			return http.DefaultTransport.RoundTrip(req)
-		})
+		apiCfg.RoundTripper = &nethttp.Transport{
```
Comment (csmarchbanks) [inline pkg/querier/correctness/runner.go]: Why the change here? Seems like it's just another wrapper but I don't think anything changed to make it needed?

File: pkg/querier/correctness/runner.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/querier/correctness/runner.go]: Ah, good catch!

File: pkg/querier/correctness/runner.go (hunk 2 above)
Comment (tomwilkie) [reply pkg/querier/correctness/runner.go]: The opentracing http client is a bit weird - you need the call to nethttp.TraceRequest _and_ the nethttp.Transport.

**Comment by tomwilkie** [comment]
Thanks!
